### DISCUSSION TITLE
A few ideas / questions for the combat system

### DISCUSSION THREAD

**Discussion by ghost** [Slow-Scallion4183/Adventure-Colussus#38] [Ideas]
Hi all - I was pointed to this page from reddit. I'm looking through the files and catching up on the discussions and had a few questions.

So, for the luck mechanic: my understanding is that luck is from [0,10], and at present, and its sole function so far is to be divided by 50, multiplied by the weapon's damage value, and added to contribute to total damage dealt. So,

Current form: Damage = (Weapon Damage) + ((Weapon Damage)*(Luck Dice-Roll / 50))

There seems to be discussion that there will eventually be NPCs to interact with; will luck contribute to having successful interactions here or elsewhere, or are there plans to introduce dedicated skills? There is a placeholder Leveling/XP system in place; what are the ideas for what Leveling up does? I also see shield and magic stats.

Did yall want RPG-style attributes, or are you sticking with damage and HP? If the former, a basic jumping off point might be to include a few attributes, e.g. strength (for melee damage), wisdom (for magic damage), dexterity (for ranged non-magic damage), and constitution (for HP). Strength could for example add a flat bonus to Weapon Damage, adding an insurance policy against bad luck & differentiate a Brawler using Hands from a Ranger using Hands. This already exists by defining how much damage Hands does directly in the Brawler / Ranger class definitions, but might get unwieldy as the number of weapons increases. Wisdom could imply that the spellcaster has a spellbook, and that you need a certain Wisdom value to be able to cast certain spells / have a better success rate. The interaction of those two skills then allows for a spellblade class that has neither excellent Strength nor Wisdom but a little of both to be versatile. 

Alternatively, characters could have weapon-specific proficiencies, wherein a weapon usually has a "damage potential" range rather than guaranteeing a flat number, but this range shrinks as you get more skillful, e.g. a Swords skill of 1 means you're hitting for [5,15] damage, but a Swords skill of 10 means you're hitting for [9,15] damage.

One more thing to think about on this front which you've probably already thought about: dice-rolling is already involved through the Luck stat. At present Luck is rolled as 1 (Luck+1)-sided die whose value is reduced by a factor of 50. The number of dice you're rolling will determine the distribution of potential damage outputs. So if you wanted Luck to be anything more intricate than a uniform distribution over [0,Luck], like having a tendency to roll high or low, just use multiple dice.

Also, for the map generation & where the character is able to move, I read that they'll use N-S-E-W to navigate through rooms and whatnot. Is there any interest in having basic map generation, e.g. a rough grid of the immediate surroundings, or is it all theatre of mind?

Cheers!

**Comment by Slow-Scallion4183** [comment]
Hello there! Thanks for starting up tis discussion :D

Ill write a detailed reply tomorrow as its getting late were I live but thanks for starting up a good discussion. Many good ideas and opportunities can start from this :D

**Comment by robotlightsyou** [comment]
I'd just like to add that all of the numbers I used when writing the basic classes were completely arbitrary. I picked round numbers that would make it easy to see if the methods were working.

As to leveling up, I figured we would attach an XP value to enemies, and you'd receive that whenever you defeated them, pretty standard. I hadn't thought about whether items or NPC encounters might also affect XP. That could be neat, but could also lead to feature creep. I think it might be wise to get the basic first encounter plotted before going too crazy on the mechanics of attributes.

RPG Style Attributes: I have no opinion as to how involved to get, I'm happy to follow on this one.

I like the dice idea. I know my general tone is for simpler, but I think dice could add a lot of variety of experience without increasing the codebase too much.

I have no strong opinions on maps v minds, again I'm happy to follow.

There's some great questions in there, thanks for bringing them up.
